Feat(front): realization page with css

tomyo002 opened the pull request.

darkweak commented on frontend/src/style.css:
```diff
@@ -2,3 +2,17 @@
+    display: flex;
+    flex: 1;
+    flex-direction: column;
+}
```
Missing NL.

darkweak reviewed (changes requested):
When I asked about screenshots, it was in the PR description, not in the code/repository.

tomyo002 committed:
fix(front):  fix language

tomyo002 committed:
feat(front):  fix NL missing

eslint

feat(front): install Owner avatar parse-prometheus-text-format  package

fix(front): display components

fix(front): install postcss to fix daisyUi

feat(front): adding navbar and footer components

feat(front): creation page component

feat(front): css

fix(front): resolve conflit

feat(front): adding picture page

fix(front): fix NL missing

tomyo002 committed:
fix(front): fix language

tomyo002 committed:
feat(front): css and colors themes

tomyo002 commented:
![Capture d’écran du 2024-04-19 14-23-40](https://github.com/tomyo002/souin-desktop/assets/132358409/d6fc3f2a-2017-476a-a7a2-96d1cbadb997)
![Capture d’écran du 2024-04-19 14-24-03](https://github.com/tomyo002/souin-desktop/assets/132358409/03b1845f-a369-42a9-95c4-07cd9a50bf3c)

darkweak commented on frontend/src/components/layout/Footer.tsx:
```diff
@@ -0,0 +1,20 @@
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path d="M22.672 15.226l-2.432.811.841 2.515c.33 1.019-.209 2.127-1.23 2.456-1.15.325-2.148-.321-2.463-1.226l-.84-2.518-5.013 1.677.84 2.517c.391 1.203-.434 2.542-1.831 2.542-.88 0-1.601-.564-1.86-1.314l-.842-2.516-2.431.809c-1.135.328-2.145-.317-2.463-1.229-.329-1.018.211-2.127 1.231-2.456l2.432-.809-1.621-4.823-2.432.808c-1.355.384-2.558-.59-2.558-1.839 0-.817.509-1.582 1.327-1.846l2.43 ...
+      </svg>
```
Useless SVG right?

darkweak commented on frontend/src/components/layout/Navbar.tsx:
```diff
@@ -0,0 +1,40 @@
+
+import { Connected } from '../molecule';
+
+type PropsNavbar = {
```
Naming.

darkweak commented on frontend/src/components/molecule/Connected.tsx:
```diff
@@ -0,0 +1,27 @@
+import { extractor } from '../../service/fetcher';
+import { AllowedIcon, Icon } from '../atomic';
+
+type PropsConnected = {
```
Naming.

darkweak commented on frontend/src/components/molecule/DisplayQuantile.tsx:
```diff
@@ -42,24 +42,26 @@ export const DisplayQuantile: React.FC<DisplayProps> = ({
         } else {
           const index = lines.findIndex(line => line['name'] === name);
-          const tmpPath = lines[index]['metrics'][0];
+          const path = lines[index].metrics[0];
```
You're accessing to one `metric` so why not naming your var `metric`?

darkweak commented on frontend/src/components/molecule/DisplayQuantile.tsx:
```diff
@@ -22,7 +22,7 @@ export const DisplayQuantile: React.FC<DisplayProps> = ({
   const [label, setlabel] = useState<Array<string>>([]);
   const [labelSum, setlabelSum] = useState<Array<string>>([]);
-  const InitializeClick = () => {
+  const ResetClick = () => {
```
Do you need this constant?

darkweak commented on frontend/src/components/page/DisplayPage.tsx:
```diff
@@ -0,0 +1,40 @@
+          title="process virtual memory bytes"
+        />
+      </div>
+      <div className="flex flex-col ">
```
```suggestion
      <div className="flex flex-col">
```
By the way, `flex flex-col` with only one item?

darkweak commented on frontend/src/components/page/DisplayPage.tsx:
```diff
@@ -0,0 +1,40 @@
+import { Navbar, Footer } from '../layout';
+import { Display, DisplayQuantile } from '../molecule';
+
+type PropsDisplayPage = {
```
Naming.

darkweak commented on frontend/tailwind.config.js:
```diff
@@ -4,4 +4,7 @@ export default {
   },
   plugins: [require('@tailwindcss/typography'), require('daisyui')],
+  daisyui: {
+    themes: ['light', 'coffee'],
```
`coffee`? Really?

darkweak commented on main.go:
```diff
@@ -19,7 +19,7 @@ func main() {
 		Title:  "Souin Desktop",
 		Width:  1024,
-		Height: 768,
+		Height: 853,
```
Arbitrary height?

darkweak reviewed (changes requested):
Take the time to apply all previous comments in your previous PRs

tomyo002 committed:
fix(front): dependance missing

tomyo002 committed:
fix(front): fix

darkweak commented:
Here is what it should look like:
![image](https://github.com/tomyo002/souin-desktop/assets/25440709/7534a276-6fb9-45e8-b3ff-0661ca111066)

I removed the theme switch and made a multiline graph so it can be reused easier.

tomyo002 committed:
feat(front): multiLineData

tomyo002 committed:
fix(front): graph color

tomyo002 committed:
feat(front): chart limit

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -29,8 +35,12 @@ export const Display: React.FC<DisplayProps> = ({ baseUrl, labels, title }) => {
+              const currentData =
+                data['data'].length >= max
+                  ? data['data'].slice(1)
+                  : data['data'];
```
```diff
            return current.map(({ data }) => {
              const currentData =
                data.length >= max
                  ? data.slice(1)
                  : data;
```

tomyo002 committed:
fix(front): fix

tomyo002 committed:
fix(front): fix name file

darkweak commented on frontend/src/service/fetcher.tsx:
```diff
@@ -19,3 +19,13 @@ export function extractor(baseUrl: string, endpoint: string) {
+export function checkHealth(baseUrl: string) {
+  return fetcher(baseUrl, '/')
+    .then(() => {
+      return 'ok';
```
Plaintext `'ok'`? Why?

darkweak commented on frontend/src/components/molecule/HealthCheck.tsx:
```diff
@@ -0,0 +1,27 @@
+  useEffect(() => {
+    const interval = setInterval(() => {
+      checkHealth(baseUrl).then(response => {
+        if (response === null) {
```
Simplify this check.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -0,0 +1,90 @@
+  useEffect(() => {
+    const interval = setInterval(() => {
+      extractor(baseUrl, '/metrics').then(lines => {
+        if (!lines) {
+          setMultiData(current => {
+            return current.map(data => {
+              const currentData =
+                data.data.length >= max ? data.data.slice(1) : data.data;
+              return {
+                data: [...currentData, 0],
+                label: data.label,
+                borderColor: data.borderColor,
+              };
+            });
+          });
+        } else {
```
duplicated code.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -0,0 +1,90 @@
+}) => {
+  const [multiData, setMultiData] = useState<ReadonlyArray<MultiDataTypes>>(
+    labels.map(label => {
+      return {
```
`return` not needed.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -0,0 +1,90 @@
+                data.data.length >= max ? data.data.slice(1) : data.data;
+              return {
+                data: [...currentData, 0],
+                label: data.label,
+                borderColor: data.borderColor,
```
Simplify.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -0,0 +1,90 @@
+      });
+
+      setlabel(current => {
+        const currentLabel = current.length >= max ? current.slice(1) : current;
```
simplify.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -0,0 +1,90 @@
+        if (!lines) {
+          setMultiData(current => {
+            return current.map(data => {
+              const currentData =
```
Useless assignment.

darkweak commented on frontend/src/components/layout/Footer.tsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Rework.

darkweak commented on frontend/src/components/molecule/Line.tsx:
```diff
@@ -1,27 +1,31 @@
-  title: string;
-  label: Array<string>;
+  data: ReadonlyArray<MultiDataTypes>;
+  labels: Array<string>;
```
`ReadonlyArray`

tomyo002 committed:
fix(front): fix footer

tomyo002 committed:
fix(front): fix

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
-            });
+        setMultiData(currents => {
+          return currents.map(({ data, label, ...rest }) => {
+            return {
```
`return` to needed.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
-              };
-            });
+        setMultiData(currents => {
+          return currents.map(({ data, label, ...rest }) => {
```
`return` to needed.

darkweak commented on frontend/src/components/molecule/MultiLineData.tsx:
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
+                        .metrics[0].value,
+                    ),
+              ],
+              label: label,
```
To simplify.

darkweak commented on frontend/src/service/fetcher.tsx:
```diff
@@ -21,11 +21,7 @@ export function extractor(baseUrl: string, endpoint: string) {
-    .catch(() => {
-      return null;
-    });
+  return fetcher(baseUrl, '/').catch(() => {
+    return null;
+  });
```
return a boolean.

darkweak commented on frontend/src/components/layout/Footer.tsx:
```diff
@@ -0,0 +1,5 @@
+export const Footer: React.FC = () => (
+  <footer className="footer-center p-4 bg-neutral text-neutral-content">
```
`footer` className is missing.

tomyo002 committed:
fix(front): fix

darkweak commented on frontend/src/components/layout/Navbar.tsx:
```diff
@@ -0,0 +1,21 @@
+export const Navbar: React.FC<NavbarProps> = ({ name, baseUrl }) => (
+  <div className="navbar bg-neutral text-neutral-content">
+    <div className="flex-1">
+      <a className="btn btn-ghost text-xl">
+        {name}: {baseUrl}
+      </a>
```
Use span instead.

darkweak commented on frontend/src/components/layout/Navbar.tsx:
```diff
@@ -0,0 +1,21 @@
+        {name}: {baseUrl}
+      </a>
+    </div>
+    <div className="flex-none">
```
Not needed.

darkweak commented on frontend/src/components/molecule/HealthCheck.tsx:
```diff
@@ -0,0 +1,23 @@
+export const HealthCheck: React.FC<HealthCheckProps> = ({ baseUrl }) => {
+  const [icon, setIcon] = useState<AllowedIcon>('x-circle');
+  useEffect(() => {
+    const interval = setInterval(() => {
+      checkHealth(baseUrl).then(response => {
+        setIcon(!response ? 'x-circle' : 'check-circle');
+      });
+    }, 1000);
```
Remove the interval, check only once.

tomyo002 committed:
fix(front): fix
